Correct details about DBT_CLOUD_ENVIRONMENT_TYPE (#6509)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/environment-variables.md

- `DBT_ENV` &mdash; This key is reserved for the dbt Cloud application and will always resolve to 'prod'. For deployment runs only.
- `DBT_CLOUD_ENVIRONMENT_NAME` &mdash; The name of the dbt Cloud environment in which `dbt` is running. 

> - `DBT_CLOUD_ENVIRONMENT_TYPE` &mdash; The type of dbt Cloud environment in which `dbt` is running. The valid values are `dev`, `staging`, or `prod`. This value can remain unset, so be sure to access it with a default (e.g. `{{env_var('DBT_CLOUD_ENVIRONMENT_TYPE', '')}}`).

Reviewer: ```suggestion
- `DBT_CLOUD_ENVIRONMENT_TYPE` &mdash; The type of dbt Cloud environment in which `dbt` is running. The valid values are `dev`, `staging`, or `prod`. It can be unset, so use a default like `{{env_var('DBT_CLOUD_ENVIRONMENT_TYPE', '')}}`.
```
